mantidproject/mantid
# Add manitdplot settings to workbench - part 1

**Description of work.**
This PR adds in mantidplot settings into workbench. This PR includes the settings in the general tab and the settings to show and hide the algorithm and interface categories.

**To test:**
Open workbench
Click `file`->`settings`
Some settings have been added to the `General` tab and a new tab has been added to allow you to show and hide algorithms and interfaces.
Change the settings and check they work as expected - compare with the nightly to check which have been added.

Note: some of the settings require you to close and reopen workbench before they take effect - this is the same behaviour as mantidplot and can't easily be avoided.

The issue also includes adding show/hide settings for functions but the functionality for this has not been added yet so this is going to be done separately.

Ref #27471

*This does not require release notes* because **they will be added in the next pr**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
